compiler/typecheck/TcErrors.hs

```diff
@@ -821,15 +823,20 @@ reportGroup mk_err ctxt cts =
             do { err <- mk_err ctxt monadFailCts
                ; reportWarning (Reason Opt_WarnMissingMonadFailInstances) err }
 
-        (_, cts') -> do { err <- mk_err ctxt cts'
+        (_, cts') -> do { damlInfo <- getEnvDaml
+                        ; let cts'' = uncurry (++) $ partition (isJust . customDamlError damlInfo) cts'
```

Comment:
I think you meant to use the commented-out `prioritizeDamlErrors`, I think it shows intent a bit better

```suggestion
                        ; let cts'' = prioritizeDamlErrors cts'
```